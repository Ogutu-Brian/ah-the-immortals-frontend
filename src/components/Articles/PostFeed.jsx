import React, { Component } from 'react';
import PropTypes from 'prop-types';
import PostItem from '../../views/Articles/PostItem';

class PostFeed extends Component {
  render() {
    const { posts } = this.props;

    return posts.map(post => <PostItem key={post.slug} post={post} />);
  }
}

PostFeed.propTypes = {
  posts: PropTypes.array.isRequired
};

export default PostFeed;
