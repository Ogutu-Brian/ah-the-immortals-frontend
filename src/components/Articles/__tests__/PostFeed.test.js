import {mount} from 'enzyme';
import React from 'react';
import PostFeed from '../PostFeed';


describe('pass post feed', () => {
  let props = [];
  const wrapper = mount(
    <PostFeed posts={props} />
  );
  it('test post feed ', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('test map function', () => {
    wrapper.find('PostFeed').find('PostItem').map(node => node.text());
    expect(wrapper).not.toBe(null);
  });
});
