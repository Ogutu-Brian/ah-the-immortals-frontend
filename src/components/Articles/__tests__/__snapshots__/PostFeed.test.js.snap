// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pass post feed test post feed  1`] = `ReactWrapper {}`;
